ffi/blitbuffer.lua

```diff
@@ -1826,6 +1826,216 @@ function BB_mt.__index:writePNGFromBGR(filename)
     C.free(cdata)
 end
 
+function BB_mt.__index:writeBMP(filename)
+    local function write_uint32(target_ptr, target_pos, data)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = data
+        return target_pos + 1
+    end
+
+    local output_channels = 3
+    local w, h = self:getWidth(), self:getHeight()
+    local stride = w * output_channels
+
+    -- we do this before all buffer alocations
+    local of, err = io.open(filename, "wb")
+    if err ~= nil then
+        return err
+    end
+
+    local bbdump
+    local data
+    if self:getType() == TYPE_BBRGB24 then
+        data = ffi.cast("unsigned char *", self.data)
+    else
+        bbdump = BB.new(w, h, TYPE_BBRGB24, nil)
+        bbdump:blitFrom(self)
+        data = ffi.cast("unsigned char *", bbdump.data)
+    end
+
+    local filesize = stride * h + 54
+    local padding = 0
+    -- update filesize, if stride is not a multiple of 4
+    if band(stride, 3) ~= 0 then
+        padding = 4 - band(stride, 3)
+        filesize = filesize + padding * h
+    end
+
+    local target_ptr = ffi.cast("unsigned char*", C.calloc(filesize, 1))
```

Comment:
I would *possibly* store the original result of calloc in a separate variable, use *that* one for the `free` call, and cast it to `uint8pt` separately.

This shouldn't matter much in practice, but you never know ;).